src/XMakeBuildEngine/Evaluation/IntrinsicFunctions.cs

```diff
@@ -306,6 +306,35 @@ internal static string GetDirectoryNameOfFileAbove(string startingDirectory, str
         }
 
         /// <summary>
+        /// Searches for a file based on the specified <see cref="IElementLocation"/>.
+        /// </summary>
+        /// <param name="file">The file to search for.</param>
+        /// <param name="elementLocation">An <see cref="IElementLocation"/>.  The directory of the <see cref="IElementLocation.File"/> is used as the starting
+        /// location of the file search.</param>
+        /// <returns>The full path of the file if it is found in a path above the <see cref="IElementLocation.File"/>.</returns>
+        internal static string GetPathOfFileAbove(string file, IElementLocation elementLocation)
+        {
+            if (String.IsNullOrWhiteSpace(elementLocation?.File))
+            {
+                ErrorUtilities.ThrowInternalError($"{nameof(elementLocation)} should not be null and it should have a file path associated with it.");
+            }
+
+            // This method does not accept a path, only a file name
+            if(file.Any(i => i.Equals(Path.DirectorySeparatorChar) || i.Equals(Path.AltDirectorySeparatorChar)))
+            {
+                ErrorUtilities.ThrowArgument("InvalidGetPathOfFileAboveParameter", file);
```

Comment:
The error is thrown and caught by the expander.  It is then wrapped as an `InvalidProjectFileException` with the location of the property function.
``` C#
catch (TargetInvocationException ex)
{
    // We ended up with something other than a function expression
    string partiallyEvaluated = GenerateStringOfMethodExecuted(_expression, objectInstance, _methodMethodName, args);
    ProjectErrorUtilities.ThrowInvalidProject(elementLocation, "InvalidFunctionPropertyExpression", partiallyEvaluated, ex.InnerException.Message.Replace("\r\n", " "));
    return null;
}
```